packages/website/docs/creating-multi-source-autocompletes.md

```diff
@@ -2,3 +2,375 @@
++ import { predefinedItemsPlugin } from './predefinedItemsPlugin';
++
+// Instantiate the autocomplete instance
+autocomplete({
+  container: '#autocomplete',
+-  plugins: [],
++  plugins: [predefinedItemsPlugin],
+  openOnFocus: true,
+});
+```
+
+Now, as soon as a user clicks on the search bar, these predefined items appear. Once they begin typing, only predefined items that contain the query remain.
+
+<AutocompleteExample
+  plugins={[predefinedItemsPlugin]}
+  openOnFocus={true}
+/>
+
+## Adding recent searches and Query Suggestions
+
+You can add recent searches and Query Suggestions using out-of-the-box plugins. Refer to the guides on [adding recent searches](using-recent-searches-plugin) and [Query Suggestions](using-query-suggestions-plugin) for more detailed explanations.
+
+### Creating a recent searches plugin
+
+Use the out-of-the-box [`createLocalStorageRecentSearchesPlugin`](createlocalstoragerecentsearchesplugin) function to create a recent searches plugin:
+
+```js
+import { createLocalStorageRecentSearchesPlugin } from '@algolia/autocomplete-plugin-recent-searches';
+
+const recentSearchesPlugin = createLocalStorageRecentSearchesPlugin({
+  key: 'RECENT_SEARCH',
+  limit: 5,
+});
+
+```
+
+The `key` can be any string and is useful to differentiate search histories if you have multiple autocompletes on one page. The `limit` defines the maximum number of recent searches to display.
+
+### Creating a Query Suggestions plugin
+
+:::note
+
+**You must have a populated [Query Suggestions](https://www.algolia.com/doc/guides/building-search-ui/ui-and-ux-patterns/query-suggestions/js/#implementing-query-suggestions) index to use this plugin.** If you don't have a Query Suggestions index yet, follow the guide on [creating a Query Suggestions index](https://www.algolia.com/doc/guides/building-search-ui/ui-and-ux-patterns/query-suggestions/how-to/creating-a-query-suggestions-index/js/).
+
+:::
+
+Use the out-of-the-box [`createQuerySuggestionsPlugin`](createQuerySuggestionsPlugin) function to create a Query Suggestions plugin. It requires an [Algolia search client](https://www.algolia.com/doc/api-client/getting-started/install/javascript/) initialized with an [Algolia application ID and API key](https://www.algolia.com/doc/guides/sending-and-managing-data/send-and-update-your-data/how-to/importing-with-the-api/#application-id) and an `indexName`. The `indexName` is the name of your [Query Suggestions](https://www.algolia.com/doc/guides/building-search-ui/ui-and-ux-patterns/query-suggestions/js/) index.
+
+```js
+import { createQuerySuggestionsPlugin } from '@algolia/autocomplete-plugin-query-suggestions';
+
+const searchClient = algoliasearch(
+  'yourAppID',
+  'yourSearchApiKey'
+);
+
+const querySuggestionsPlugin = createQuerySuggestionsPlugin({
+  searchClient,
+  indexName: 'yourQuerySuggestionsIndexName',
+  },
+});
+```
+
+### Aligning Query Suggestions to items from different sources
```

Comment:
Is "aligning" the right word?

Reply:
It might not be, but let me double check I understand the goal of this.

The idea is to have the `hitsPerPage` for both recentSearches + querySuggestions = 5, right?

For boolean parameters, e.g. click analytics, it would mean that both plugins use the same value, right?

Reply:
It means that a value of 10 will display `nQS = 10 - nRS`. "Balancing" might fit better?

Reply:
I don't think balancing is quite right either. "Coordinating" might work.

Is the hitsPerPage query parameter the only intended use case for this, or could it be applied to other query parameters as well?